iree/compiler/Dialect/HAL/Target/test/smoketest.mlir

```diff
@@ -1,4 +1,4 @@
-// RUN: iree-opt -split-input-file -iree-hal-transformation-pipeline -iree-hal-target-backends=vulkan-spirv %s | IreeFileCheck %s -check-prefix=VKSPV
+// RUN: [[ $IREE_VULKAN_DISABLE == 1 ]] || (iree-opt -split-input-file -iree-hal-transformation-pipeline -iree-hal-target-backends=vulkan-spirv %s | IreeFileCheck %s -check-prefix=VKSPV)
```

Comment:
This is going to disable this test on CI that can't run the Vulkan driver.  Previously vulkan_disable meant that the *driver* was not available. We're ramping up Vulkan bot coverage, but even then I don't think we should conflate not having the driver with not having the backend.

Reply:
Good point, though see my other comment about the ambiguity with this simple filtering mechanism.

Reply:
Moved this test under `HAL/Target/VulkanSPIRV`, added filtering at the build level, and reverted the `IREE_VULKAN_DISABLE` check. This test should still be run now, and it's placed using the same scheme as the other smoketest.mlir files for other HAL targets.